packages/auth/plugin/src/ios/openUrlFix.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  ConfigPlugin,
+  withAppDelegate,
+  withInfoPlist,
+  ExportedConfigWithProps,
+  WarningAggregator,
+} from '@expo/config-plugins';
+import type { ExpoConfig } from '@expo/config/build/Config.types';
+import type { AppDelegateProjectFile } from '@expo/config-plugins/build/ios/Paths';
+import type { InfoPlist } from '@expo/config-plugins/build/ios/IosConfig.types';
+import { mergeContents } from '@expo/config-plugins/build/utils/generateCode';
+import { PluginConfigType } from '../pluginConfig';
+
+export const withIosCaptchaOpenUrlFix: ConfigPlugin<PluginConfigType> = (
+  config: ExpoConfig,
+  props?: PluginConfigType,
+) => {
+  // check configuration
+  if (!shouldApplyIosOpenUrlFix({ config, props })) {
+    return config;
+  }
+
+  // check that swizzling is enabled; otherwise patch must be customized and applied manually
+  withInfoPlist(config, config => {
+    ensureFirebaseSwizzlingIsEnabled(config);
+    return config;
+  });
+
+  // apply patch
+  return withAppDelegate(config, config => {
+    return withOpenUrlFixForAppDelegate({ config, props });
+  });
+};
+
+// Interpret the plugin config to determine whether this fix should be applied
+export function shouldApplyIosOpenUrlFix({
+  config,
+  props,
+}: {
+  config: ExpoConfig;
+  props?: PluginConfigType;
+}): boolean {
+  const flag = props?.ios?.captchaOpenUrlFix;
+  if (flag === undefined || flag === 'default') {
+    // by default, apply the fix whenever 'expo-router' is detected in the same project
+    return isPluginEnabled(config, 'expo-router');
+  } else if (flag === true || flag === false) {
+    const isEnabled: boolean = flag;
+    return isEnabled;
+  } else {
+    throw new Error(`Unexpected value for 'captchaOpenUrlFix' config option`);
+  }
+}
+
+export function withOpenUrlFixForAppDelegate({
+  config,
+  props,
+}: {
+  config: ExportedConfigWithProps<AppDelegateProjectFile>;
+  props?: PluginConfigType;
+}) {
+  const { language, contents } = config.modResults;
+
+  if (['objc', 'objcpp'].includes(language)) {
+    const newContents = modifyObjcAppDelegate(contents);
+    if (newContents === null) {
+      if (props?.ios?.captchaOpenUrlFix === true) {
+        throw new Error("Failed to apply iOS openURL fix because no 'openURL' method was found");
+      } else {
+        WarningAggregator.addWarningIOS(
+          '@react-native-firebase/auth',
+          "Skipping iOS openURL fix because no 'openURL' method was found",
+        );
+        return config;
+      }
+    } else {
+      return {
```

Comment:
I don't know what the typical Expo plugin user experience is like, so this may be something no one wants.
However, if we're modifying the AppDelegate for expo-router users by default, would people be interested in seeing the equivalent of an "info" printout (definitely not warning, just info) that says something like `@react-native-firebase/auth Expo config plugin modified iOS AppDelegate openURL method to ignore firebaseauth reCAPTCHA redirect URLs."

My thought is it could be juuussst enough of a breadcrumb for people to search the plugin for the message and learn what's going on if they have problems with it ever.

Or it might be something no expo plugin does, for reasons (perhaps "quiet" is encouraged and noise is strongly discouraged?).

Any answer is fine, "implementor's choice" meaning ignore this idea or do something about it at your full option as you'll have more experience here than me either way